tawazi/_dag/dag.py

```diff
@@ -422,17 +422,24 @@ def setup(
                 and the root nodes is logical.
         """
         # 1. if target_nodes is not provided run all setup ExecNodes
-        target_ids = (
+        target_nodes = (
             self.graph_ids.setup_nodes
             if target_nodes is None
-            else self._sanitize_nodes_alias(target_nodes)
+            else self.get_multiple_nodes_aliases(target_nodes)
         )
 
         # 2. the leaves_ids that the user wants to execute
-        #  however they might contain non setup nodes... so we should extract all the nodes ids
-        #  that must be run in order to run the target_nodes ExecNodes
-        graph = self.make_subgraph(
-            target_nodes=target_ids, exclude_nodes=exclude_nodes, root_nodes=root_nodes
+        exclude_nodes = (
+            self.get_multiple_nodes_aliases(exclude_nodes)
+            if exclude_nodes is not None
+            else exclude_nodes
+        )
```

Comment:
```suggestion
        if exclude_nodes is not None:
            exclude_nodes = self.get_multiple_nodes_aliases(exclude_nodes)
```
I find this easier to understand

Reply:
done